### PR TITLE
[Robust Speech Event] Add guides

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/research_projects/xls_r/eval.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
```
Comment (patrickvonplaten) [inline examples/research_projects/xls_r/eval.py:1]: @anton-l @Narsil - this should be the official `eval.py` script that people should take as a format for the event. Would be great if you could review it :-)

File: examples/research_projects/xls_r/README.md [hunk 2]
```diff
@@ -393,3 +470,99 @@ error rate of 27% as can be seen on the automatically generated [model card](htt
...
+- how to add the built *n-gram* the speech recognition system for seamless decoding
+
+Our previously trained model - [xls-r-300m-sv](https://huggingface.co/hf-test/xls-r-300m-sv) - enjoys a 30% word error rate reduction after 
+having added an n-gram. As shown in the example of the blog post, we strongly advise participants to upload all files required for combining 
+the *n-gram* with a trained speech recognition model directly into the same model repository.
+
+## Evaluation
+
+```bash
+./eval.py --model_id hf-test/xls-r-300m-sv --dataset mozilla-foundation/common_voice_7_0 --config sv-SE --split test --log_outputs
```
Comment (patrickvonplaten) [inline examples/research_projects/xls_r/README.md:505]: works like a charm and gives good results.

File: examples/research_projects/xls_r/README.md [hunk 3]
```diff
@@ -393,3 +470,99 @@ error rate of 27% as can be seen on the automatically generated [model card](htt
...
+```
+
+works, but
+
+```bash 
+./eval.py --model_id hf-test/xls-r-300m-sv --dataset speech-recognition-community-internal/tedx_manual_dev_test --config sv --split validation --chunk_length_s 5.0 --stride_length_s 1.0 --log_outputs
```
Comment (patrickvonplaten) [inline examples/research_projects/xls_r/README.md:511]: This is not supported yet. It's doing pipeline with chunking + language model integration. Would be great if we could make it possible to do this for the sprint. It should not be too difficult actually. We can just do the same trick with the output logits vectors instead of the argmax output ids (as done for no LM) and then decode one huge logits matrix

File: examples/research_projects/xls_r/README.md [hunk 4]
```diff
@@ -393,3 +470,99 @@ error rate of 27% as can be seen on the automatically generated [model card](htt
 The above-chosen hyperparameters probably work quite well on a range of different 
 datasets and languages, but are by no means optimal. It is up to you to find a good set of 
 hyperparameters.
+
+
+## How to finetune with OVH cloud
+
+TODO(Anton)
```
Comment (anton-l) [inline examples/research_projects/xls_r/README.md:477]:
```suggestion
## How to finetune with OVH cloud

For a more detailed guide on setting up OVHcloud please watch this video: TODO

### Creating an OVHCloud account
*TIP*: If you haven't created a project on OVHcloud yet, make sure you've received your GPU voucher code *beforehand*, 
so that you can skip entering the credit card information.
1. If you're a US citizen, create an account via [OVHcloud.CA](https://ovhcloud.ca/). 
If you're from anywhere else in the world, create an account via [OVHcloud.COM](https://ovhcloud.com/).
2. Once logged in, click `Public Cloud` from the top menu and then click `Create your first OVH Public Cloud project`. 
Then enter a project name (e.g. "huggingface"), enter your voucher code, and click `Continue` -> `Create my project`.
*Note: if you see a request for credit card details during the last step, and you can't skip it, then your voucher code 
is invalid. Please report it to the [#ovh-support](https://discord.gg/p4qqDV3M) channel on Discord.*

### Setting up an AI notebook
1. Go to the `Public Cloud` page and select `Project Management` -> `Users & Roles` from the menu on the left. 
2. Click `+ Add user`. Write a user description (e.g. `AI Trainer`), and select an `AI Training Operator` user role. 
Click `Confirm`.
3. Write down the *username* and *password* (at the top of the screen) somewhere. They will be needed during step 7.
4. Select `AI & Machine Learning` -> `AI Training` from the menu on the left. 
Click `+ Launch a new job` on the AI Training page.
5. On the `Launch a new job` page:
   * In `1. Choose a region` select a region closest to you.
   * In `2. Enter the Docker image` select `Custom image` -> `baaastijn/ovh_huggingface`.
   * You can skip steps `3.` and `4.` if you will be using the Hugging Face Hub to store the models after training.
   * In `5. Configure your job` select **1** `GPU`.
   * Validate the info and Create the job.
6. On the `AI Training Jobs` screen wait until the job's status changes from `Pending` to `Running`.
7. Click `HTTP Access` and log in with the AI training user you've created earlier. 
Once logged in, you can close the page and click `HTTP Access` to launch a JupyterLab notebook.
8. Awesome, now you have a free GPU-enabled Jupyter instance!

**Note**: If you're an experienced Docker user, feel free to create a custom docker image with all of the needed packages 
like the one in step 5. The Dockerfile for it is available here: 
[baaastijn/Dockerimages](https://github.com/baaastijn/Dockerimages/tree/main/Hugginface_challenge_speech).
Once you've built your own image, push it to https://hub.docker.com/ and select it during the OVHcloud job creation.
```

File: examples/research_projects/xls_r/eval.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    chars_to_ignore_regex = '[,?.!\-\;\:\"“%‘”�—’…–]'  # noqa: W605 IMPORTANT: this should correspond to the chars that were ignored during training
+
+    text = re.sub(chars_to_ignore_regex, "", text.lower())
+
+    # In addition, we can normalize the target text, e.g. removing new lines characters etc...
+    # note that order is important here!
+    token_sequences_to_ignore = ["\n\n", "\n", "   ", "  "]
+
+    for t in token_sequences_to_ignore:
+        text = " ".join(text.split(t))
```
Comment (anton-l) [inline examples/research_projects/xls_r/eval.py:62]: These can be safely replaced by just `text = " ".join(text.split())`

File: examples/research_projects/xls_r/eval.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    )
+    parser.add_argument(
+        "--split", type=str, required=True, help="Split of the dataset. *E.g.* `'test'`"
+    )
+    parser.add_argument(
+        "--chunk_length_s", type=float, default=None, help="Chunk length in seconds. Defaults to 5 seconds."
+    )
+    parser.add_argument(
+        "--stride_length_s", type=float, default=None, help="Stride of the audio chunks. Defaults to 1 second."
+    )
```
Comment (anton-l) [inline examples/research_projects/xls_r/eval.py:120]:
```suggestion
    parser.add_argument(
        "--chunk_length_s", type=float, default=5, help="Chunk length in seconds. Defaults to 5 seconds."
    )
    parser.add_argument(
        "--stride_length_s", type=float, default=1, help="Stride of the audio chunks. Defaults to 1 second."
    )
```

File: examples/research_projects/xls_r/eval.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+from datasets import load_dataset, load_metric, Audio, Dataset
+from transformers import pipeline, AutoFeatureExtractor
+import re
+import argparse
```
Comment (anton-l) [inline examples/research_projects/xls_r/eval.py:5]:
```suggestion
import argparse
import unicodedata
```

File: examples/research_projects/xls_r/eval.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+            result.map(write_to_file, with_indices=True)
+
+
+def normalize_text(text: str) -> str:
+    """ DO ADAPT FOR YOUR USE CASE. this function normalizes the target text. """
+
+    chars_to_ignore_regex = '[,?.!\-\;\:\"“%‘”�—’…–]'  # noqa: W605 IMPORTANT: this should correspond to the chars that were ignored during training
+
+    text = re.sub(chars_to_ignore_regex, "", text.lower())
```
Comment (anton-l) [inline examples/research_projects/xls_r/eval.py:55]:
```suggestion
    text = text.lower()
    # normalize non-standard (stylized) unicode characters
    text = unicodedata.normalize('NFKC', text) 
    # remove punctuation
    text = re.sub(chars_to_ignore_regex, "", text)
```

**Review by anton-l — APPROVED** [review]

File: examples/research_projects/xls_r/eval.py (hunk 6 above)
Comment (patrickvonplaten) [reply examples/research_projects/xls_r/eval.py:120]: yeah will change the text here. Still don't want people to default to chunking if not necessary

File: examples/research_projects/xls_r/README.md [hunk 9]
```diff
@@ -1,58 +1,81 @@
-# Speech recognition community week - version 2 🤗
```
Comment (osanseviero) [inline examples/research_projects/xls_r/README.md:1]: Should the directory be changed from xls_r to speech_challenge?

File: examples/research_projects/xls_r/README.md [hunk 10]
```diff
@@ -1,58 +1,81 @@
...
 
-If necessary and available, free access to a V100 32 GB GPU will kindly be provided by the [OVH team](https://us.ovhcloud.com/) 🚀.
+The goal of this event is to build **robust**, **real-world** speech recognition (ASR) systems in as many languages as possible 🌏🌍🌎.
+If necessary and available, free access to a V100 32 GB GPU will kindly be provided by the [OVH could team](https://us.ovhcloud.com/) 🚀.
+This document summarizes all the relevant information required for the speech community event 📋.
 
-This document summarizes all the relevant information required for the speech community event📋.
-
-Don't forget to sign up [here](TODO: Create google from)🤗.
+To sign-up, please see [this forum post](https://discuss.huggingface.co/t/open-to-the-community-robust-speech-recognition-challenge/13614) 🤗.
```
Comment (osanseviero) [inline examples/research_projects/xls_r/README.md:9]:
```suggestion
To sign-up, please see [this forum post](https://discuss.huggingface.co/t/open-to-the-community-robust-speech-recognition-challenge/13614) 🤗. Please make sure to:
- Read it in detail
- Fill the google form
- Join our Discord server in the #join-sprint channel.
```

**Review by osanseviero — APPROVED** [review]
